Feature/metadataset

cmhughes opened the pull request:
These two commits implement the `\metadataset` command for each of the four OU class files: `ouab.cls`, `ouexam.cls`, `ouicma.cls` and `outn.cls`; this feature was first proposed in https://github.com/rbrignall/OU-SUPPS/issues/25

This new feature does not break or change anything existing, but simply provides a different way of configuring the meta data for a document.

I've updated `Documentation/ou-supps.tex` to account for this, but, for example, you can now use:

`abtemplate.tex`
-
```
\metadataset{faculty=Science Technology Engineering and Mathematics, 
            module code=MST125,
            module title=Title of Module,
            document title=AB title,
            document subtitle=AB subtitle,
            presentation=2017B,
            copy right year=1999,
            supp no=3.14159,
            version no=1.12,
            multiple choice option text=choices,
            no cut off date,
            instructions=some specific instructions,
            }
```

You'll see analogous examples for each of the other class files. 

The *second* commit adds the following summary table, detailing which keys are available for each of the four document classes:

![image](https://cloud.githubusercontent.com/assets/2224480/26157806/0fd1a926-3b12-11e7-9ed7-784fecd4e245.png)

I hope this sounds reasonable and useful -- of course, any and all feedback is very welcome!
Best wishes
Chris

cmhughes committed:
Implementation of metadataset which accepts key=value pairs to set meta data for the document

cmhughes committed:
Summary table of keys available to metadataset

rbrignall commented:
Hi Chris,

Looks good. Just a one minor comment: throughout, "copy right" should probably be one word "copyright".

rbrignall commented:
One other comment: ou-supps.pdf is not currently merging cleanly -- not sure if there's anything simple you can do at your end? Otherwise, I'll just force the issue at this end.

cmhughes committed:
Changed copy right year to copyright year

cmhughes commented:
Many thanks, Robert; I've changed `copy right year` to `copyright year`. The `ou-supps.pdf` also merges cleanly now; I believe this was because my other pull request also modified the file.
